src/azure-cli/azure/cli/command_modules/keyvault/commands.py

```diff
@@ -73,21 +74,23 @@ def load_command_table(self, _):
                                    'the soft deleted state. Please see the following documentation for additional '
                                    'guidance.\n'
                                    'https://docs.microsoft.com/en-us/azure/key-vault/general/soft-delete-overview'))
-        g.command('purge', 'purge_deleted')
-        g.custom_command('set-policy', 'set_policy')
-        g.custom_command('delete-policy', 'delete_policy')
+        g.command('purge', 'begin_purge_deleted', supports_no_wait=True)
+        g.custom_command('set-policy', 'set_policy', supports_no_wait=True)
+        g.custom_command('delete-policy', 'delete_policy', supports_no_wait=True)
         g.command('list-deleted', 'list_deleted')
         g.generic_update_command(
             'update', setter_name='update_keyvault_setter', setter_type=kv_vaults_custom,
             custom_func_name='update_keyvault',
-            doc_string_source='azure.mgmt.keyvault.v' + mgmt_api_version + '.models#VaultProperties')
+            doc_string_source='azure.mgmt.keyvault.v' + mgmt_api_version + '.models#VaultProperties',
+            supports_no_wait=True)
+        g.wait_command('wait')
 
     with self.command_group('keyvault network-rule',
                             kv_vaults_sdk,
                             min_api='2018-02-14',
                             client_factory=keyvault_client_vaults_factory) as g:
-        g.custom_command('add', 'add_network_rule')
-        g.custom_command('remove', 'remove_network_rule')
+        g.custom_command('add', 'add_network_rule', supports_no_wait=True)
+        g.custom_command('remove', 'remove_network_rule', supports_no_wait=True)
```

Comment:
in these scenarios, is there a wait command to wait util condition satisfied?

Reply:
@Juliehzl Good catch. Added.